File: CS/DXApplication3/Form1.cs
using System;
using System.Data;

namespace DXApplication3
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            customGrid1.DataSource = CreateTable(42);
            customGridView1.OptionsSelection.MultiSelect = true;
        }
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, i % 4, DateTime.Now.AddDays(i) });
            return tbl;
        }
    }
}

File: VB/DXApplication3/Form1.vb
Imports System.Data

Namespace DXApplication3

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
            customGrid1.DataSource = CreateTable(42)
            customGridView1.OptionsSelection.MultiSelect = True
        End Sub

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Number", GetType(Integer))
            tbl.Columns.Add("Date", GetType(Date))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i, i Mod 4, Date.Now.AddDays(i)})
            Next

            Return tbl
        End Function
    End Class
End Namespace
